shared/src/main/scala/org/atnos/eff/Interpret.scala

```diff
@@ -233,6 +233,13 @@ trait Interpret {
     })
   }
 
+  /**
+   * For a single effect T log every value of that effect
+   */
+  def trace[R, F[_], A](eff: Eff[R, A])(implicit m: MemberInOut[F, R]): Eff[Fx.prepend[Writer[F[_], ?], R], A] =
```

Comment:
I am not sure about this signature. I think it would be more general to require an implicit `w: Writer[F[_]] |= R`.